ceph/rados/core_workflows.py

```diff
@@ -1746,6 +1751,292 @@ def get_daemon_status(self, daemon_type, daemon_id) -> tuple:
+            daemon_info = {}
+            for entry in data:
+                daemon_type = entry["daemon_type"]
+                daemon_name = entry["daemon_name"]
+                hostname = entry["hostname"]
+                if daemon_type not in daemon_info:
+                    daemon_info[daemon_type] = {}
+                daemon_info[daemon_type][daemon_name] = hostname
+            return daemon_info
+
+        # Collect daemon information from both JSON structures
+        orch_ps_daemons = collect_daemon_info(orch_ps_data)
+        orch_ps_new_daemons = collect_daemon_info(orch_ps_new_data)
+
+        # Compare daemons and print the results
+        all_match = True
+        for daemon_type, daemons in orch_ps_daemons.items():
+            if daemon_type in orch_ps_new_daemons:
+                for daemon_name, hostname in daemons.items():
+                    if daemon_name in orch_ps_new_daemons[daemon_type]:
+                        new_hostname = orch_ps_new_daemons[daemon_type][daemon_name]
+                        is_same = hostname == new_hostname
+                        if not is_same:
+                            all_match = False
+                            log.error(f"Daemon Type: {daemon_type}")
+                            log.error(f"  Daemon Name: {daemon_name}")
+                            log.error(f"    Old Hostname: {hostname}")
+                            log.error(f"    New Hostname: {new_hostname}")
+                            log.error(f"    Same Hostname: {is_same}")
+                    else:
+                        all_match = False
+                        log.error(f"Daemon Type: {daemon_type}")
+                        log.error(f"  Daemon Name: {daemon_name}")
+                        log.error(f"    Old Hostname: {hostname}")
+                        log.error("    New Hostname: Not found")
+                        log.error("    Same Hostname: False")
+            else:
+                all_match = False
+                log.error(f"Daemon Type: {daemon_type} not found post tests")
+
+        # Check for missing daemons in orch_ps_new
+        for daemon_type, daemons in orch_ps_daemons.items():
+            if daemon_type not in orch_ps_new_daemons:
+                all_match = False
+                log.error(f"Daemon Type: {daemon_type} is missing post tests")
+            else:
+                for daemon_name in daemons:
+                    if daemon_name not in orch_ps_new_daemons[daemon_type]:
+                        all_match = False
+                        log.error(
+                            f"Daemon Name: {daemon_name} in "
+                            f"Daemon Type: {daemon_type} is missing in orch ps post tests."
+                        )
+
+        # Check for new daemons added in orch_ps post upgrade
+        for daemon_type, daemons in orch_ps_new_daemons.items():
+            if daemon_type not in orch_ps_daemons:
+                all_match = False
```

Comment:
According to me, failure is not required if a new daemon is added to the cluster

Reply:
Yes, We are not passing fail if there are new daemons found post tests. that is just a comment that we are printing in the logs.